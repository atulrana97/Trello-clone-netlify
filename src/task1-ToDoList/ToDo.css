.todo-input {
  margin-right: 0.3rem;
  margin-left: 0.3rem;
}
.final-container {
  display: none;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
.items p input[type='text'] {
  font-family: 'Poppins', sans-serif;
  font-weight: bolder;
  font-size: 1rem;
  text-shadow: 0.05rem 0.05rem rgb(202, 126, 126);
  margin: 0.5rem;
  text-align: start;
}
.items-list {
  padding-left: 0;
  margin-left: 0;
}
.items {
  margin-top: 0.5rem;
  border: solid 0.3rem black;
  box-shadow: 0.125rem 0.125rem rgb(212, 96, 96);
  border-radius: 0.7rem;
  display: flex;
  list-style: none;
  flex-direction: row;
  justify-content: space-around;
}
.submit-button {
  padding: 0.5rem;
  height: 0.7rem;
  border-radius: 0.3rem;
}
#heading {
  font-family: 'Poppins', sans-serif;
  font-size: 4rem;
  text-shadow: 0.125rem 0.125rem rgb(212, 96, 96);
}
.todo-container {
  margin-top: 2rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 0.2rem;
  border-radius: 0.4rem;
}
.delete-button {
  color: white;
  background-color: rgb(212, 96, 96);
  font-family: 'Poppins', sans-serif;
  font-weight: bolder;
  font-size: 1rem;
  border-radius: 0.7rem;
}

.items p input {
  background-color: transparent;
  border: 0;
}
.items p input:focus {
  outline: none;
}
