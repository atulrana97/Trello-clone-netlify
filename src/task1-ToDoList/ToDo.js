import React, { Component } from 'react'
import './ToDo.css'

class ToDo extends Component {
  constructor() {
    super()
    this.state = {
      toDoArray: [],
      currentItem: {
        key: '',
        value: ''
      }
    }
    this.listContainer = React.createRef()
  }

  addItems = (e) => {
    e.preventDefault()
    const newItem = this.state.currentItem
    if (newItem.value !== '') {
      const items = [...this.state.toDoArray, newItem]
      this.setState({
        toDoArray: items,
        currentItem: {
          key: '',
          value: ''
        }
      })
      this.listContainer.current.style.border = 'solid 0.3rem black'
      this.listContainer.current.style.boxShadow =
        '0.125rem 0.125rem rgb(212, 96, 96)'
    }
  }

  handleInputChange = (e) => {
    this.setState({
      currentItem: {
        key: Date.now(),
        value: e.target.value
      }
    })
  }

  deleteItem = (key) => {
    const filteredItems = this.state.toDoArray.filter(
      (item) => item.key !== key
    )
    this.setState({
      toDoArray: filteredItems
    })
    if (this.state.toDoArray.length === 1) {
      this.listContainer.current.style.border = 'solid 0rem black'
      this.listContainer.current.style.boxShadow = '0rem 0rem rgb(212, 96, 96)'
    }
  }

  updateToDoArray = (event, key) => {
    console.log('items:' + this.state.toDoArray)
    const items = this.state.toDoArray
    items.map((item) => {
      item.value = item.key === key ? event.target.value : item.value
      return item
    })
    this.setState({
      toDoArray: items
    })
  }

  render() {
    return (
      <div className="final-container">
        <h1 id="heading" data-testid="toDoTitle">
          To Do (
          <span data-testid="toDoCount">{this.state.toDoArray.length}</span>)
        </h1>
        <form onSubmit={this.addItems} data-testid="formTestId">
          <input
            className="todo-input"
            type="text"
            onChange={this.handleInputChange}
            data-testid="inputTextField"
            value={this.state.currentItem.value}
          ></input>
          <input
            data-testid="submitBtn"
            type="submit"
            value="Add Item"
            name="AddItem"
            className="submit-btn"
          />
        </form>
        <div
          className="todo-container"
          id="toDoContainer"
          ref={this.listContainer}
        >
          <ul className="items-list">
            {this.state.toDoArray.map((ele) => (
              <li className="items" key={ele.key}>
                <p>
                  <input
                    data-testid="addedItem"
                    type="text"
                    id={ele.key}
                    value={ele.value}
                    onChange={(e) => {
                      this.updateToDoArray(e, ele.key)
                    }}
                  ></input>
                </p>
                <button
                  data-testid="deleteBtn"
                  type="button"
                  className="delete-button"
                  onClick={() => {
                    this.deleteItem(ele.key)
                  }}
                >
                  Remove
                </button>
              </li>
            ))}
          </ul>
        </div>
      </div>
    )
  }
}
export default ToDo
