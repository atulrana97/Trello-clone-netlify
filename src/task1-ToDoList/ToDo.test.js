import React from 'react'
import {
  render,
  fireEvent,
  getByTestId,
  getAllByTestId
} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'
import ToDo from './ToDo'

let container = null
beforeEach(() => {
  container = render(<ToDo />).container
})
afterEach(() => {
  container = null
})
describe('Test for ToDo', () => {
  it('renders correctly todo component', () => {
    const tree = renderer.create(<ToDo />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should show title of toDo', () => {
    expect(getByTestId(container, 'toDoTitle')).toBeInTheDocument()
  })
  it('should show form of toDo', () => {
    expect(getByTestId(container, 'formTestId')).toBeInTheDocument()
  })
  it('should check if the add items buttons adds the task', async () => {
    const inputText = getByTestId(container, 'inputTextField')
    fireEvent.change(inputText, { target: { value: 'atul' } })
    const button = getByTestId(container, 'submitBtn')
    fireEvent.click(button)
    const toDoCount = await getByTestId(container, 'toDoCount').textContent
    const addedText = await getByTestId(container, 'addedItem').value
    expect(addedText).toBe('atul')
    expect(toDoCount).toBe('1')
  })
  it('should check if the delete items buttons deletes the task card', async () => {
    const inputText = getByTestId(container, 'inputTextField')
    fireEvent.change(inputText, { target: { value: 'atul' } })
    const button = getByTestId(container, 'submitBtn')
    fireEvent.click(button)
    const deleteButton = await getByTestId(container, 'deleteBtn')
    fireEvent.click(deleteButton)
    const toDoCount = await getByTestId(container, 'toDoCount').textContent
    expect(toDoCount).toBe('0')
  })
  it('should check if the tasks are editable', async () => {
    const cardOne = await getByTestId(container, 'inputTextField')
    fireEvent.change(cardOne, { target: { value: 'rana' } })
    const submitButton = await getByTestId(container, 'submitBtn')
    fireEvent.click(submitButton)
    const cardTwo = await getByTestId(container, 'inputTextField')
    fireEvent.change(cardTwo, { target: { value: 'rana' } })
    fireEvent.click(submitButton)
    const inputText = await getAllByTestId(container, 'addedItem')
    fireEvent.change(inputText[0], {
      target: { value: 'atul' }
    })
    expect(inputText[0].value).toBe('atul')
  })
  it('should check if the empty task is not pushed in toDo List', async () => {
    const addItemTextBox = await getByTestId(container, 'inputTextField')
    fireEvent.change(addItemTextBox, { target: { value: '' } })
    const submitButton = await getByTestId(container, 'submitBtn')
    fireEvent.click(submitButton)
    const toDoCount = await getByTestId(container, 'toDoCount').textContent
    expect(toDoCount).not.toBe('1')
  })
  it('should check if correct task is deleted on deleting the task', async () => {
    const cardOne = await getByTestId(container, 'inputTextField')
    fireEvent.change(cardOne, { target: { value: 'rana' } })
    const submitButton = await getByTestId(container, 'submitBtn')
    fireEvent.click(submitButton)
    const cardTwo = await getByTestId(container, 'inputTextField')
    fireEvent.change(cardTwo, { target: { value: 'Atul' } })
    fireEvent.click(submitButton)
    const deleteButton = await getAllByTestId(container, 'deleteBtn')
    fireEvent.click(deleteButton[1])
    expect(cardTwo.value).not.toBe('Atul')
  })
})
