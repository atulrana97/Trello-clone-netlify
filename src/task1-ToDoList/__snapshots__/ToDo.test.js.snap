// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for ToDo renders correctly todo component 1`] = `
<div
  className="final-container"
>
  <h1
    data-testid="toDoTitle"
    id="heading"
  >
    To Do (
    <span
      data-testid="toDoCount"
    >
      0
    </span>
    )
  </h1>
  <form
    data-testid="formTestId"
    onSubmit={[Function]}
  >
    <input
      className="todo-input"
      data-testid="inputTextField"
      onChange={[Function]}
      type="text"
      value=""
    />
    <input
      className="submit-btn"
      data-testid="submitBtn"
      name="AddItem"
      type="submit"
      value="Add Item"
    />
  </form>
  <div
    className="todo-container"
    id="toDoContainer"
  >
    <ul
      className="items-list"
    />
  </div>
</div>
`;
